### PR TITLE
Support active directory pwd authentication

### PR DESCRIPTION
**Pull request by aaomidi**
Adds support for a new authentication model - username/password authentication with AAD.

### CODE REVIEW

**Comment by aaomidi** [comment]
https://github.com/microsoft/sqltoolsservice/pull/899

File: extensions/mssql/package.nls.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
 	"mssql.connectionOptions.authType.categoryValues.sqlLogin": "SQL Login",
 	"mssql.connectionOptions.authType.categoryValues.integrated": "Windows Authentication",
 	"mssql.connectionOptions.authType.categoryValues.azureMFA": "Azure Active Directory - Universal with MFA support",
+	"mssql.connectionOptions.authType.categoryValues.activeDirectoryPassword": "Active Directory - Password",
```
Comment (kburtram) [inline extensions/mssql/package.nls.json]: I guess this is the "best" name we have for this option? 😄

Comment (aaomidi) [reply extensions/mssql/package.nls.json]: I really want a better name! Have any suggestions?

**Comment by hummerpj** [comment]
Hi all,

Any plans to merge this soon?  Would really love to see this land in ADS. 

Thank you!

**Comment by kburtram** [comment]
@hummerpj I think there were a couple issues we needed to track down before we can merge this. 
 @aaomidi can provide better details when he's back in the office next week.  I'm not sure if this will make Feb release, or be held off for March release.

**Comment by arpishah2** [comment]
Hello,
Seems the changes are approved. Can this be merged before March. Have been waiting for this to get merged. Thanks!

**Comment by aaomidi** [comment]
This is on my plate on things to tackle. @abist has been working on getting this working in the mssql vscode extension. I'll work with them to get this onto ADS as well.

**Comment by aaomidi** [comment]
Unfortunately this is going to be a no-fix: https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/482

Looks like we're just going to have to stick to azure authentication.

**Comment by aaomidi** [comment]
Reverting: https://github.com/microsoft/sqltoolsservice/pull/915

**Comment by hummerpj** [comment]
I read through most of that thread and I am not sure I understand why it can't be supported.

Can't it be updated to MSAL.NET and supported through that?

It's disappointing that other applications on OSX are capable of accepting active directory username/password yet the main Microsoft applications can not.  

Unfortunately, we are stuck using AD to access our DBs given the enterprise environment we are in. I presume, based on all the threads on this topic, that many others are stuck in a similar situation.

I appreciate the follow-up on this.

**Pull request closed without merging**